File: Python/lab17/Consultant.py
from ConsultantAbstract import ConsultantAbstract
from Order import Order


class Consultant(ConsultantAbstract):
    def __init__(self, mediator, name, role):
        super().__init__(mediator, name, role)

    def send_order(self, order: Order):
        print(f"Consultant: {self.name}, role: {self.role}, send order: {order}")
        self.mediator.send_order(order, self)

    def get_order(self, order: Order):
        print(f"Consultant: {self.name}, role: {self.role}, get order: {order}")

File: Java/lab17/src/main/java/org/example/Consultant.java
package org.example;

public class Consultant extends ConsultantAbstract{
    public Consultant(IMediator mediator, String name, String role) {
        super(mediator, name, role);
    }

    @Override
    public void sendOrder(Order order) {
        System.out.println("Consultant: "+name+", role: "+role+", send order: "+order);
        mediator.sendOrder(order,this);
    }

    @Override
    public void getOrder(Order order) {
        System.out.println("Consultant: "+name+", role: "+role+", get order: "+order);
    }
}
